### PR TITLE
[NO-ISSUE] Use apache-release profile when doing a release build

### PR DESCRIPTION
**Pull request by baldimir**
Adds gpg setup and apache-release Maven profile to the release build.

Similar (can be merged separately) PRs:
- [kogito-runtimes](https://github.com/apache/incubator-kie-kogito-runtimes/pull/3530)
- [kogito-apps](https://github.com/apache/incubator-kie-kogito-apps/pull/2062)
- [optaplanner](https://github.com/apache/incubator-kie-optaplanner/pull/3093)

### CODE REVIEW

File: .ci/jenkins/Jenkinsfile.deploy [hunk 1]
```diff
@@ -121,14 +121,29 @@ pipeline {
                         } else {
                             installOrDeploy = 'install'
                         }
+                        def mavenCommand = getMavenCommand()
+                           .withOptions(env.DROOLS_BUILD_MVN_OPTS ? [ env.DROOLS_BUILD_MVN_OPTS ] : [])
+                           .withOptions(env.BUILD_MVN_OPTS_CURRENT ? [ env.BUILD_MVN_OPTS_CURRENT ] : [])
+                           .withProperty('maven.test.failure.ignore', true)
+                           .skipTests(params.SKIP_TESTS)
+                           .withSettingsXmlFile(MAVEN_SETTINGS_FILE)
```
Comment (jstastny-cz) [inline .ci/jenkins/Jenkinsfile.deploy]: this needs to be in `configFileProvider` section for MAVEN_SETTINGS_FILE to exist (it's tmp scoped to the defining configFileProvider block).
I'd say extend the configFileProvider block above the def mavenCommand.

**Review by jstastny-cz — CHANGES_REQUESTED** [review]

File: .ci/jenkins/Jenkinsfile.deploy (hunk 1 above)
Comment (jstastny-cz) [reply .ci/jenkins/Jenkinsfile.deploy]: Or move mavenCommand.withSettingsXmlFile into the configFileProvider + run section.

Comment (baldimir) [reply .ci/jenkins/Jenkinsfile.deploy]: Fixed.

**Review by jstastny-cz — APPROVED** [review]
LGTM.

**Pull request merged**